src/graph.py

```diff
@@ -52,25 +53,27 @@ def edges(self):
         list_key_value = self._dict.items()
         list_edges = []
         for pair in list_key_value:
-            for node in pair[1]:
-                list_edges.append((pair[0], node),)
+            for tup in pair[1]:
+                list_edges.append((pair[0], tup[0], tup[1]),)
         return list_edges
 
     def del_node(self, n):
         """Delete a node from graph."""
-        if (n) in self._dict:
+        if n in self._dict:
             del self._dict[n]
             for key in self._dict:
-                if (n) in self._dict[key]:
-                    node_value = self._dict[key]
-                    node_value.remove(n)
+                for tup in self._dict[key]:
```

Comment:
Be aware that this is an O(n^2) operation, since `list.remove` is an O(n) operation which you are calling for each item in the edges list.